### PR TITLE
docs: add openapi documentation for app schemas

### PR DESCRIPTION
**Pull request by pyphilia**
ref #1528

### CODE REVIEW

File: src/services/item/plugins/app/appData/schemas.ts [hunk 1]
```diff
@@ -49,10 +55,16 @@ export const create = {
...
   response: {
     [StatusCodes.OK]: appDataSchemaRef,
+    '4xx': errorSchemaRef,
   },
 };
 
 export const updateOne = {
+  operationId: 'updateAppData',
+  tags: ['app', 'app-data'],
+  summary: 'Update app data',
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/schemas.ts:65]:
```suggestion
  summary: 'Update an app data',
```

File: src/services/item/plugins/app/appData/schemas.ts [hunk 2]
```diff
@@ -62,20 +74,33 @@ export const updateOne = {
...
   response: {
     [StatusCodes.OK]: appDataSchemaRef,
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
 export const deleteOne = {
+  operationId: 'deleteAppData',
+  tags: ['app', 'app-data'],
+  summary: 'Delete app data',
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/schemas.ts:84]:
```suggestion
  summary: 'Delete an app data',
```

File: src/services/item/plugins/app/appData/schemas.ts [hunk 3]
```diff
@@ -62,20 +74,33 @@ export const updateOne = {
...
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
 export const getForOne = {
+  operationId: 'getAppDataForApp',
+  tags: ['app', 'app-data'],
+  summary: 'Get all app data of an app',
+  description:
+    'Get app data saved for an app, depending on the permission of the user and the data visibility.',
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/schemas.ts:102]: We have the possibility to specify a `type` filter to only get app-data of the specified type.

File: src/services/item/plugins/app/appData/schemas.ts [hunk 4]
```diff
@@ -87,6 +112,7 @@ export const getForOne = {
     additionalProperties: false,
   },
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/schemas.ts:113]: The Querystring object is not typed using typebox, should it be added ? And documented ?

File: src/services/item/plugins/app/appSetting/schemas.ts [hunk 5]
```diff
@@ -22,13 +23,18 @@ export const appSettingSchemaRef = registerSchemaAsRef(
       updatedAt: customType.DateTime(),
     },
     {
-      // Schema Options
+      description: 'Settings saved for an app.',
```
Comment (spaenleh) [inline src/services/item/plugins/app/appSetting/schemas.ts:26]:
```suggestion
      description: 'Settings for an app.',
```

File: src/services/item/plugins/app/appSetting/schemas.ts [hunk 6]
```diff
@@ -59,20 +71,32 @@ export const updateOne = {
...
     [StatusCodes.OK]: appSettingSchemaRef,
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
 export const getForOne = {
+  operationId: 'getAppSettingsForApp',
+  tags: ['app', 'app-setting'],
+  summary: 'Get all settings of an app',
+  description: 'Get all settings saved for an app.',
```
Comment (spaenleh) [inline src/services/item/plugins/app/appSetting/schemas.ts:98]: Same comment here, it is possible to select only settings with a specific name.

File: src/services/item/plugins/app/appSetting/schemas.ts [hunk 7]
```diff
@@ -59,20 +71,32 @@ export const updateOne = {
...
+
   params: Type.Object({
     itemId: customType.UUID(),
   }),
```
Comment (spaenleh) [inline src/services/item/plugins/app/appSetting/schemas.ts:102]: Use the typebox object to type the querystring parameter ?

File: src/services/item/plugins/app/schemas.ts [hunk 8]
```diff
@@ -73,24 +26,77 @@ export const generateToken = {
...
+          additionalProperties: false,
+        },
+      ),
+    ),
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
 export const getMostUsed = {
+  operationId: 'getMostUsedApp',
```
Comment (spaenleh) [inline src/services/item/plugins/app/schemas.ts:81]:
```suggestion
  operationId: 'getMostUsedApps',
```

File: src/services/item/plugins/app/schemas.ts [hunk 9]
```diff
@@ -73,24 +26,77 @@ export const generateToken = {
...
+  tags: ['app'],
+  summary: 'Get informations about most used apps',
+  description: 'Get informations about most used apps',
```
Comment (spaenleh) [inline src/services/item/plugins/app/schemas.ts:84]:
```suggestion
  summary: 'Get the user's most used apps',
  description: 'Get a list of the apps the user has used the most to ease the addition of new apps.',
```

File: src/services/item/plugins/app/schemas.ts [hunk 10]
```diff
@@ -73,24 +26,77 @@ export const generateToken = {
...
-    [StatusCodes.OK]: Type.Array(mostUsedAppSchemaRef),
+    [StatusCodes.OK]: Type.Array(
+      Type.Object(
+        {
+          name: Type.String(),
+          url: Type.String(),
+          count: Type.Number(),
+        },
+        {
+          description: 'Stats of most used apps',
```
Comment (spaenleh) [inline src/services/item/plugins/app/schemas.ts:95]:
```suggestion
          description: 'An app regularly used by a member',
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Thank you for the effort on documenting these endpoints. I am not sure how precise we should be with the descriptions versus summaries. Let me know what you think.

File: src/services/item/plugins/app/appData/schemas.ts (hunk 3 above)
Comment (pyphilia) [reply src/services/item/plugins/app/appData/schemas.ts:102]: I added the info in the description of the qs directly.

File: src/services/item/plugins/app/appData/test/index.test.ts [hunk 11]
```diff
@@ -261,6 +261,7 @@ describe('App Data Tests', () => {
           url: `${APP_ITEMS_PREFIX}/${v4()}/app-data`,
           payload,
         });
+        console.log(response);
```
Comment (ReidyT) [inline src/services/item/plugins/app/appData/test/index.test.ts:264]:
```suggestion
```

**Review by ReidyT — APPROVED** [review]
Thanks for this PR, LGTM ☕️

**Review by spaenleh — APPROVED** [review]
Well done !

**Pull request merged**